snuba_sdk/mql_visitor.py

```diff
@@ -103,32 +112,41 @@ def _visit_query(self, query: Timeseries | Formula | None) -> Mapping[str, str]:
 
         return self.timeseries_visitor.visit(query)
 
-    def _visit_start(self, start: datetime | None) -> datetime:
+    def _visit_start(self, start: datetime | None) -> str:
         if start is None:
             raise InvalidMetricsQueryError("MetricQuery.start must not be None")
 
-        return start
+        return start.isoformat()
 
-    def _visit_end(self, end: datetime | None) -> datetime:
+    def _visit_end(self, end: datetime | None) -> str:
         if end is None:
             raise InvalidMetricsQueryError("MetricQuery.end must not be None")
 
-        return end
+        return end.isoformat()
 
-    def _visit_rollup(self, rollup: Rollup | None) -> Rollup:
+    def _visit_rollup(self, rollup: Rollup | None) -> dict[str, str | int | None]:
         if rollup is None:
             raise InvalidMetricsQueryError("MetricQuery.rollup must not be None")
 
-        return rollup
+        return self.rollup_visitor.visit(rollup)
 
-    def _visit_scope(self, scope: MetricsScope | None) -> MetricsScope:
+    def _visit_scope(self, scope: MetricsScope | None) -> dict[str, str | list[int]]:
         if scope is None:
             raise InvalidMetricsQueryError("MetricQuery.scope must not be None")
 
-        return scope
+        return self.scope_visitor.visit(scope)
 
-    def _visit_limit(self, limit: Limit | None) -> Limit | None:
-        return limit
+    def _visit_limit(self, limit: Limit | None) -> int | None:
+        return limit.limit if limit is not None else None
```

Comment:
If we're going to change this to allow `None`, we need to update the parser to reflect this: https://github.com/getsentry/snuba/pull/5119/files#diff-602995071cc68736a6c6f5eecb5672ec85419d9a80b9e3aba7ac963b098c70b6R694-R751